File: .skills/openclaw-skills/skills/itonlyforfun-ai/bitget-poolx-monitor/billing.py
#!/usr/bin/env python3
"""
SkillPay.me 收费接口
"""
import requests

BILLING_API_URL = "https://skillpay.me"
BILLING_API_KEY = "sk_0c57911690182c45404f945f2908a9e2e32ed448055895ac02227856eaf226ad"
SKILL_ID = "6a3b1843-5266-40c6-8f0c-408512bb6f43"

def check_balance(user_id):
    """查询用户余额"""
    url = f"{BILLING_API_URL}/api/v1/billing/balance"
    headers = {"X-API-Key": BILLING_API_KEY}
    
    try:
        resp = requests.get(f"{url}?user_id={user_id}", headers=headers, timeout=10)
        data = resp.json()
        return data.get("balance", 0)
    except Exception as e:
        print(f"查询余额错误: {e}")
        return 0

def charge_user(user_id, amount=0.001):
    """
    扣费（每次调用）
    amount: USDT 数量，默认 0.001
    """
    url = f"{BILLING_API_URL}/api/v1/billing/charge"
    headers = {
        "X-API-Key": BILLING_API_KEY,
        "Content-Type": "application/json"
    }
    data = {
        "user_id": user_id,
        "skill_id": SKILL_ID,
        "amount": amount
    }
    
    try:
        resp = requests.post(url, json=data, headers=headers, timeout=10)
        result = resp.json()
        
        if result.get("success"):
            return {"ok": True, "balance": result.get("balance", 0)}
        else:
            return {
                "ok": False, 
                "balance": result.get("balance", 0),
                "payment_url": result.get("payment_url", "")
            }
    except Exception as e:
        print(f"扣费错误: {e}")
        return {"ok": False, "error": str(e)}

def get_payment_link(user_id, amount=10):
    """生成充值链接"""
    url = f"{BILLING_API_URL}/api/v1/billing/payment-link"
    headers = {
        "X-API-Key": BILLING_API_KEY,
        "Content-Type": "application/json"
    }
    data = {"user_id": user_id, "amount": amount}
    
    try:
        resp = requests.post(url, json=data, headers=headers, timeout=10)
        result = resp.json()
        return result.get("payment_url", "")
    except Exception as e:
        print(f"生成充值链接错误: {e}")
        return ""

# 使用示例
if __name__ == "__main__":
    user = "test_user"
    
    # 1. 查询余额
    balance = check_balance(user)
    print(f"用户余额: {balance} USDT")
    
    # 2. 扣费
    result = charge_user(user, 0.001)
    if result.get("ok"):
        print("✅ 扣费成功，执行功能")
    else:
        print(f"❌ 扣费失败: {result}")
        if result.get("payment_url"):
            print(f"充值链接: {result['payment_url']}")

File: .skills/openclaw-skills/skills/squallsol/session-password/scripts/billing.js
/**
 * SkillPay Billing Integration for Session Password
 * 
 * Pricing Model:
 * - Buyout: 29 USDT (one-time, lifetime access)
 * - Per-call: 0.9 USDT (if not purchased)
 * 
 * Skill ID: 6387f3fa-cceb-4a60-afd4-fc1f4eb2d7c3
 * API: BNB Chain USDT payment
 */

const BILLING_API_URL = 'https://skillpay.me';
const BILLING_API_KEY = 'sk_20cb7dd70ceaa244814a9e9a0e990a2781fab10e163fd2f9499f5bb1e7a6eaef';
const SKILL_ID = '6387f3fa-cceb-4a60-afd4-fc1f4eb2d7c3';

// Pricing
const PRICE_PER_CALL = 0.01; // USDT - per call

// ═══════════════════════════════════════════════════
// Core Billing Functions
// ═══════════════════════════════════════════════════

// ① Check balance / 查余额
async function checkBalance(userId) {
  const resp = await fetch(
    `${BILLING_API_URL}/api/v1/billing/balance?user_id=${userId}`,
    { headers: { 'X-API-Key': BILLING_API_KEY } }
  );
  const data = await resp.json();
  return data.balance; // USDT amount
}

// ② Check if user has purchased (buyout) / 检查是否已买断
async function checkPurchased(userId) {
  const resp = await fetch(
    `${BILLING_API_URL}/api/v1/billing/access?user_id=${userId}&skill_id=${SKILL_ID}`,
    { headers: { 'X-API-Key': BILLING_API_KEY } }
  );
  const data = await resp.json();
  return data.has_access === true; // true = already purchased
}

// ③ Charge per call / 每次调用扣费
async function chargeUser(userId, amount = PRICE_PER_CALL) {
  const resp = await fetch(`${BILLING_API_URL}/api/v1/billing/charge`, {
    method: 'POST',
    headers: {
      'X-API-Key': BILLING_API_KEY,
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      user_id: userId,
      skill_id: SKILL_ID,
      amount: amount,
    }),
  });
  const data = await resp.json();

  if (data.success) {
    return { ok: true, balance: data.balance };
  }

  // Insufficient balance → get payment link
  return { ok: false, balance: data.balance, paymentUrl: data.payment_url };
}



// ⑤ Generate payment link / 生成充值链接
async function getPaymentLink(userId, amount = 10) {
  const resp = await fetch(`${BILLING_API_URL}/api/v1/billing/payment-link`, {
    method: 'POST',
    headers: {
      'X-API-Key': BILLING_API_KEY,
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ user_id: userId, amount }),
  });
  const data = await resp.json();
  return data.payment_url; // BNB Chain USDT payment link
}

// ═══════════════════════════════════════════════════
// Main Handler - Smart Billing Logic
// ═══════════════════════════════════════════════════

/**
 * Handle billing for skill execution
 * 
 * Logic:
 * 1. Charge 0.01 USDT per call
 * 2. Insufficient balance → Return payment link
 * 
 * @param {string} userId - User identifier
 * @returns {Promise<{allowed: boolean, message: string, paymentUrl?: string}>}
 */
async function handleBilling(userId) {
  try {
    // Charge per call
    const result = await chargeUser(userId, PRICE_PER_CALL);
    
    if (result.ok) {
      return { 
        allowed: true, 
        message: `✅ Charged $${PRICE_PER_CALL} USDT. Session Password activated.` 
      };
    }

    // Insufficient balance
    const paymentUrl = await getPaymentLink(userId, 5);
    
    return { 
      allowed: false,
      balance: result.balance,
      paymentUrl,
      message: `❌ Insufficient balance.\n\n📊 Pricing:\n• Per call: $${PRICE_PER_CALL} USDT\n\n💳 Recharge: ${paymentUrl}`
    };
  } catch (error) {
    console.error('Billing error:', error);
    // Graceful degradation - allow access if billing fails
    return { 
      allowed: true, 
      message: '⚠️ Billing system unavailable. Access granted.' 
    };
  }
}

/**
 * Get pricing info for display
 */
function getPricingInfo() {
  return {
    perCall: PRICE_PER_CALL,
    currency: 'USDT',
    chain: 'BNB Chain'
  };
}

module.exports = {
  checkBalance,
  checkPurchased,
  chargeUser,
  getPaymentLink,
  handleBilling,
  getPricingInfo,
  SKILL_ID,
  PRICE_PER_CALL
};
